go/test/endtoend/recovery/pitr/shardedpitr_test.go

```diff
@@ -0,0 +1,539 @@
+	require.NoError(t, err)
+
+	if isSlow {
+		time.Sleep(1 * time.Second)
+	}
+}
+
+func createRestoreKeyspace(t *testing.T, timeToRecover, restoreKeyspaceName string) {
+	output, err := clusterInstance.VtctlclientProcess.ExecuteCommandWithOutput("CreateKeyspace",
+		"-keyspace_type=SNAPSHOT", "-base_keyspace="+keyspaceName,
+		"-snapshot_time", timeToRecover, restoreKeyspaceName)
+	log.Info(output)
+	require.NoError(t, err)
+}
+
+func testTabletRecovery(t *testing.T, binlogServer *binLogServer, lookupTimeout, restoreKeyspaceName, shardName, expectedRows string) {
+	recoveryTablet := clusterInstance.NewVttabletInstance("replica", 0, cell)
+	launchRecoveryTablet(t, recoveryTablet, binlogServer, lookupTimeout, restoreKeyspaceName, shardName)
+
+	sqlRes, err := recoveryTablet.VttabletProcess.QueryTablet(getCountID, keyspaceName, true)
+	require.NoError(t, err)
+	assert.Equal(t, expectedRows, sqlRes.Rows[0][0].String())
+
+	defer recoveryTablet.MysqlctlProcess.Stop()
+	defer recoveryTablet.VttabletProcess.TearDown()
+}
+
+func launchRecoveryTablet(t *testing.T, tablet *cluster.Vttablet, binlogServer *binLogServer, lookupTimeout, restoreKeyspaceName, shardName string) {
+	tablet.MysqlctlProcess = *cluster.MysqlCtlProcessInstance(tablet.TabletUID, tablet.MySQLPort, clusterInstance.TmpDirectory)
+	err := tablet.MysqlctlProcess.Start()
+	require.NoError(t, err)
+
+	tablet.VttabletProcess = cluster.VttabletProcessInstance(tablet.HTTPPort,
+		tablet.GrpcPort,
+		tablet.TabletUID,
+		clusterInstance.Cell,
+		shardName,
+		keyspaceName,
+		clusterInstance.VtctldProcess.Port,
+		tablet.Type,
+		clusterInstance.TopoProcess.Port,
+		clusterInstance.Hostname,
+		clusterInstance.TmpDirectory,
+		clusterInstance.VtTabletExtraArgs,
+		clusterInstance.EnableSemiSync)
+	tablet.Alias = tablet.VttabletProcess.TabletPath
+	tablet.VttabletProcess.SupportsBackup = true
+	tablet.VttabletProcess.Keyspace = restoreKeyspaceName
+	tablet.VttabletProcess.EnableSemiSync = true
+	tablet.VttabletProcess.ExtraArgs = []string{
+		"-disable_active_reparents",
+		"-enable_replication_reporter=false",
+		"-init_db_name_override", dbName,
+		"-init_tablet_type", "replica",
+		"-init_keyspace", restoreKeyspaceName,
+		"-init_shard", shardName,
+		"-binlog_host", binlogServer.hostname,
+		"-binlog_port", fmt.Sprintf("%d", binlogServer.port),
+		"-binlog_user", binlogServer.username,
+		"-pitr_gtid_lookup_timeout", lookupTimeout,
+		"-vreplication_healthcheck_topology_refresh", "1s",
+		"-vreplication_healthcheck_retry_delay", "1s",
+		"-vreplication_tablet_type", "replica",
+		"-vreplication_retry_delay", "1s",
+		"-degraded_threshold", "5s",
+		"-lock_tables_timeout", "5s",
+		"-watch_replication_stream",
+		"-serving_state_grace_period", "1s",
+	}
+	tablet.VttabletProcess.ServingStatus = ""
+
+	err = tablet.VttabletProcess.Setup()
+	require.NoError(t, err)
+
+	tablet.VttabletProcess.WaitForTabletTypesForTimeout([]string{"SERVING"}, 20*time.Second)
+	require.Nil(t, err)
```

Comment:
we don't need this as that is checked already in 2 line above.

Reply:
Removed